weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -108,6 +119,11 @@ class CallEndCHInsertable(BaseModel):
     input_refs: typing.List[str] = []  # sadly, this is required
     output_refs: typing.List[str]
 
+    _project_id_v = field_validator("project_id")(validation.project_id_validator)
```

Comment:
would be nice if we can somehow dry this out, maybe with a RequiredField base model or something?